### DISCUSSION TITLE
Announcing initial work on v0.5!

### DISCUSSION THREAD

**Discussion by geofmureithi** [apalis-dev/apalis#235] [Announcements]
A lot of thinking went into this version.

There were specific issues that I wanted to fix from a maintainer perspective:
- Handling of features was not great and led to breaking docs among other issues.
- Controlled polling. Allows the user to control the polling process.
- Worker notify: Allows the worker to say when it is ready to consume a job.
- Fine-grained control using custom Future and Stream implementations

There is still some work remaining, but the tricky part is over.
Please see PR #234 for reference and give feedback if possible!

**Comment by thevivekshukla** [comment]
thanks for all the hard work man, I recently discovered your project and it seems fitting for my use case.

**Comment by alexted** [comment]
@geofmureithi I express my great gratitude to you for your hard work! This is a truly great implementation of Distributed Task Queue. 
Please tell me, do you know about the [Celery](https://docs.celeryq.dev/en/stable/index.html) project? Were you inspired by it while creating your product? If yes, do you plan to add the similar functionalities in your product to establish parity or just because some ideas seem useful to you and you would like to have it in Apalis too?

**Reply by geofmureithi** [reply]
Hey Alex, yes! I love celery, but I am trying to build something better or at-least similar._ I do believe we have most features offered by celery. Do you have anything specific in mind?

**Reply by alexted** [reply]
If I have any specific ideas, I'll let you know!
Thank you so much again for your tremendous efforts - you really do have the best product available!!
